platform/linuxbsd/wayland/display_server_wayland.cpp

```diff
@@ -1142,6 +1142,13 @@ void DisplayServerWayland::cursor_set_custom_image(const Ref<Resource> &p_cursor
 	}
 }
 
+bool DisplayServerWayland::get_swap_cancel_ok() {
+	String current_desk = OS::get_singleton()->get_environment("XDG_CURRENT_DESKTOP").to_lower();
+	String session_desk = OS::get_singleton()->get_environment("XDG_SESSION_DESKTOP").to_lower();
+
+	return (current_desk.contains("kde") || session_desk.contains("kde") || current_desk.contains("lxqt") || session_desk.contains("lxqt"));
+}
```

Comment:
I would check it on startup and cache the result to a member variable, so we don't read from the environment every time this is queried. It's not used _a ton_ but there's various editor plugins which seem to call this more than once.